### ISSUE TITLE
[APT-18] [Bug] Aptos CLI stopped working without the internet / hangs a lot with slow internet

### ISSUE THREAD

**Issue by mkurnikov** [aptos-labs/aptos-core#5909]
# 🐛 Bug

Aptos CLI hits the internet on every invocation now, very hard to use on a connection with high latency. 

## To reproduce

Try to disable your WI-FI and then run the `aptos move test` command, it now fails with 
```
{
  "Error": "Unexpected error: Unable to resolve packages for package 'PACKAGE_NAME'"
}
```

With slow internet, it now hangs in the beginning of the call, with nothing relevant in `RUST_LOG=debug`. I haven't looked at the code, but I remember hearing something about updating dependencies to the latest commit at the branch, maybe it's because of that. 

I'm not sure how to fix the behavior with a slow connection. At least allow it to proceed okay if there's no internet. 

UPD. While writing the issue, I noticed the new flag
```
        --skip-fetch-latest-git-deps
            Skip pulling the latest git dependencies
            
            If you don't have a network connection, the compiler may fail due to no ability to pull
            git dependencies.  This will allow overriding this for local development.
```
which should fix the issue. Could you make it more prominent? Maybe check whether there's a connection, and then enable the flag automatically if there's none?
Or at least fail with a more obvious error in case there's no connection, mentioning the flag.  

UPD2. The flag doesn't really work for me though. 

<sub>[APT-18](https://linear.app/aptoslabs/issue/APT-18/[bug]-aptos-cli-stopped-working-without-the-internet-hangs-a-lot-with)</sub>

**Comment by gregnazario** [comment]
What's the version of teh CLI?

There is that flag, which should fix the issue.  It gets more complicated on checking of there's a connection and then handling it, but I agree a more obvious error should take place.

This was to mainly fix the issue that people had out of date dependencies, which was an entirely different issue, but slow / no internet would cause issues here.

**Comment by mkurnikov** [comment]
aptos 1.0.2, pre-build release. 

```
$ aptos move test --skip-fetch-latest-git-deps
{
  "Error": "Unexpected error: Unable to resolve packages for package 'Liquidswap'"
}

```

It's how it behaves now without internet connection.

**Comment by vivekascoder** [comment]
Yeah, I've faced this issue many times, shouldn't it cache the dependency of the project and reuse it without internet?

**Comment by banool** [comment]
We have a similar issue probably with submitting analytics info, see https://github.com/aptos-labs/aptos-core/issues/8113.

**Comment by 0xjinn** [comment]
@mkurnikov Hi, I tried `aptos move test` without internet connection, it works on my side. 

Can you show me a step by step on how to repro this? Thanks!

**Comment by mkurnikov** [comment]
It works on my side with `--skip-fetch-latest-git-deps` now. 

Fails without though, I assume it tries to poll git apis at some point. I've added `--skip-fetch-latest-git-deps` parameter for my test runs, it's acceptable compromise. 
I tried to implement "detect Internet connection" functionality inside the IDE, it's not an ideal solution as it adds additional delay for ICMP call. So it's understandable from Aptos CLI side to not implement that. 

I'm closing the issue for now.
